tests/client/net_handler_client.c

```diff
@@ -163,26 +163,38 @@ static int recv_tcp_data(robot_desc_field_t client, int tcp_fd) {
         return -1;
     }
 
-    // unpack the message
-    if ((msg = text__unpack(NULL, len, buf)) == NULL) {
-        fprintf(tcp_output_fp, "Error unpacking incoming message from %s\n", client_str);
-    }
-
-    // print the incoming message
-    if (msg_type == LOG_MSG) {
-        for (int i = 0; i < msg->n_payload; i++) {
-            fprintf(tcp_output_fp, "%s", msg->payload[i]);
+    if (msg_type == TIME_STAMP_MSG) {
+        TimeStamps* time_stamp_msg = time_stamps__unpack(NULL, len, buf);
+        if (time_stamp_msg == NULL) {
+            log_printf(ERROR, "recv_new_msg: Cannot unpack time_stamp msg");
+        }
+        uint64_t final_timestamp = millis();
+        printf("First Dawn Timestamp: %lu\n", time_stamp_msg->dawn_timestamp);
+        printf("Runtime Timestamp: %lu\n", time_stamp_msg->runtime_timestamp);
+        printf("Final Dawn Timestamp: %lu\n", final_timestamp);
+        printf("Round Dawn trip: %lu\n", final_timestamp - time_stamp_msg->dawn_timestamp);
+        time_stamps__free_unpacked(time_stamp_msg, NULL);
+    } else {
+        if ((msg = text__unpack(NULL, len, buf)) == NULL) {
+            fprintf(tcp_output_fp, "Error unpacking incoming message from %s\n", client_str);
         }
-    } else if (msg_type == CHALLENGE_DATA_MSG) {
-        for (int i = 0; i < msg->n_payload; i++) {
-            fprintf(tcp_output_fp, "Challenge %d result: %s\n", i, msg->payload[i]);
+        // print the incoming message
+        if (msg_type == LOG_MSG) {
+            // unpack the message
+            for (int i = 0; i < msg->n_payload; i++) {
+                fprintf(tcp_output_fp, "%s", msg->payload[i]);
+            }
+        } else if (msg_type == CHALLENGE_DATA_MSG) {
+            for (int i = 0; i < msg->n_payload; i++) {
+                fprintf(tcp_output_fp, "Challenge %d result: %s\n", i, msg->payload[i]);
+            }
         }
+        fflush(tcp_output_fp);
+        text__free_unpacked(msg, NULL);
```

Comment:
Seems like the `if` cases are incorrect. Is the following what was intended?
```
if (msg_type == TIME_STAMP_MSG) {
...
} else if (msg_type == LOG_MSG) {
...
} else if (msg_type == CHALLENGE_DATA_MSG) {
...
} else {
// Unhandled case
}
```

Reply:
This would probably make more sense; in the past we would only be receiving text messages from Runtime (`LOG` and `CHALLENGE_DATA`) so I had just a simple unpack as text protobuf and then depending on the message type, print it out a certain way. But now that we're receiving three different types of messages, might be easier to do `if` - `else-if` - `else` like you suggested above.

Reply:
Its not that the if statements are wrong, its more so I just grouped all the ones relating to text unpacking into one condition and then had additional checks in that one condition to see the msg_type. Functionality wise it's still the same as before. I made it that way to just have one place where text is unpacked then we can see which condition to enter.

Reply:
I can modify it to fit the `if`-`else if`- `else` structure if you want it to be cleaner though

Reply:
Oh I see! I think it's just more intuitive to me if it was a `switch`/`case` altogether with a case for each message type, but this is fine!

Reply:
Yea I do see your point and agree, I'll change it and push it in a bit